internal/hcsoci/resources_wcow.go

```diff
@@ -171,6 +175,36 @@ func setupMounts(ctx context.Context, coi *createOptionsInternal, r *resources.R
 					return errors.Wrapf(err, "adding SCSI EVD mount failed %+v", mount)
 				}
 				r.Add(scsiMount)
+			} else if strings.HasPrefix(mount.Source, "sandbox://") {
+				// Mounts that map to a path in the UVM are specified with a 'sandbox://' prefix.
+				//
+				// Example: sandbox:///a/dirInUvm destination:C:\\dirInContainer.
+
+				// First convert to a path in the sandboxmounts path itself.
+				sandboxPath := convertToWCOWSandboxMountPath(mount.Source)
+
+				// Now we need to exec a process in the vm that will make these directories as theres
+				// no functionality in the Windows gcs to create an arbitrary directory.
+				//
+				// Create the directory, but also run dir afterwards regardless of if mkdir succeeded to handle the case where the directory already exists
+				// e.g. from a previous container specifying the same mount (and thus creating the same directory).
+				cmd := cmd.CommandContext(
+					ctx,
+					coi.HostingSystem,
+					"cmd",
+					"/c",
+					"mkdir",
+					sandboxPath,
```

Comment:
I think we are vulnerable to Command Injection attacks here. If the `mount.Source` is provided as `sandbox://foo & del /s C:\\Windows\\System32` it would silently delete the system32 directory. We should do some path validation to avoid that. 
I am surprised to see that there are no path validation libraries available for Golang. We probably will have to write our own function here.

Reply:
Good point, I'll figure something out!

Reply:
Well actually, I'm pretty sure whatever is in `sandboxPath` will just get treated as one argument to mkdir. So it will try and make a directory with whatever `sandoxPath` contains so I don't think we're in the red here. For example:

`sandbox:///foo & mkdir hiya howdy`

will just result in a directory being created named "foo & mkdir hiya howdy". If I use your example, as you might expect, you get "The filename, directory name, or volume label syntax is incorrect."

Reply:
@ambarve Let me know if you see any other concerns with this

Reply:
Ping

Reply:
@ambarve ping again

Reply:
Yeah, this looks good then.